src/VisualStudio/Core/Def/Implementation/LanguageClient/VisualStudioInProcLanguageServer.cs

```diff
@@ -0,0 +1,409 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.LanguageServer;
+using Microsoft.CodeAnalysis.LanguageServer.Api;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Roslyn.Utilities;
+using StreamJsonRpc;
+
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.LanguageClient
+{
+    /// <summary>
+    /// Implementation of <see cref="InProcLanguageServer"/> that also supports
+    /// VS LSP extension methods.
+    /// </summary>
+    internal class VisualStudioInProcLanguageServer : InProcLanguageServer
```

Comment:
This is the other half of the old InProcLanguageServer that contains the VS specific extensions and publish diagnostics (to be deprecated).  Everything was copied from the old class.